build.gradle

```diff
@@ -872,6 +873,16 @@ project(':clients') {
     }
   }
 
+
+  task jarTest (type: Jar, dependsOn: testClasses) {
+    from sourceSets.test.output
+    classifier = 'test'
+  }
+
+  artifacts {
+    testOutput jarTest
```

Comment:
Interesting. Does this create a new artifact? I think maybe we should go back to your previous solution that is limited to `copyDependantTestLibs` with a comment explaining why it's needed. We can consider separately whether we should move away from `testCompile project(':clients').sourceSets.test.output` in favour of the more complete solution across the board (we use test dependencies in multiple modules).

Reply:
There's also the option of just using `https://plugins.gradle.org/plugin/com.github.hauner.jarTest/1.0`, but it's best to discuss that in a separate PR.

Reply:
makes sense. reverted the last commit. Will handle this in separate PR.
For now, added a comment to explain the need for previous solution.